### PR TITLE
Create groups prior to user creation

### LINKED ISSUE #164: Missing group prior to user creation
The user creation task failed if groups are not created:
```
$ ansible-playbook users.yml -i hosts.old -u ubuntu -e @users-config.yml
|...]
TASK [Add users] *******************************************************************************************************************************************************************************
Monday 22 June 2020  22:03:21 +0200 (0:00:02.859)       0:00:02.911 *********** 
failed: [51.68.113.239] (item={'name': 'claire2', 'password': 'claire'}) => {"ansible_loop_var": "item", "changed": false, "item": {"name": "claire2", "password": "claire"}, "msg": "Group claire2 does not exist"}
failed: [51.68.113.239] (item={'name': 'sandrine2', 'password': 'sandrine'}) => {"ansible_loop_var": "item", "changed": false, "item": {"name": "sandrine2", "password": "sandrine"}, "msg": "Group sandrine2 does not exist"}

[...]
```

If passed to ansible, user groups need to exist.

### PR DESCRIPTION
**Pull request by pierrepo**
- [x] Add / update the documentation
- Should fix #164

### CODE REVIEW

**Comment by pierrepo** [comment]
Maybe the `Adding a user to groups with Ansible` section in the `Managing UNIX groups` doc should be modified since user group must exists before being associated to a user.

**Comment by jtpio** [comment]
Thanks @pierrepo for this!

File: docs/install/users.rst [hunk 1]
```diff
@@ -1,33 +1,43 @@
...
 ------------------------
 
-The ``ansible/`` directory contains a ``users.yml`` playbook that makes it easier to create new users on the host in batches.
+The ``ansible/`` directory contains a ``users.yml`` playbook that makes it easier to create new users and user groups on the host in batches.
 
 First you need to create a new ``users-config.yml`` with the following content:
 
 .. code-block:: yaml
 
+    user_groups:
```
Comment (jtpio) [inline docs/install/users.rst:23]: Should it be `user_groups` or `groups`?

(no strong feeling, it's more about succinctness)

Comment (pierrepo) [reply docs/install/users.rst:23]: I tried but `groups` is a variable already [taken](https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html) by Ansible. It took me sometime to realize this ;-)

**Comment by pierrepo** [comment]
I guess we should also refactor the documentation in the `Adding a user to groups with Ansible` section.

File: docs/permissions/groups.rst [hunk 2]
```diff
@@ -1,72 +1,53 @@
...
-Adding a user to groups with Ansible
-------------------------------------
+    usermod -a -G test alice
 
-Alternatively, groups can be defined in the users playbook, as mentioned in :ref:`install/users-playbook`.
-
-To add users to specific groups, edit ``users-config.yml`` to add the ``groups`` field:
-
-.. code-block:: yaml
+To remove a user ``alice`` to the ``test`` group:
```
Comment (jtpio) [inline docs/permissions/groups.rst:46]:
```suggestion
To remove the user ``alice`` from the ``test`` group:
```

**Pull request merged**